sage.py

```diff
@@ -237,80 +177,19 @@ def load_data(path_to_alerts, filtering_window):
-                           'Misc activity': 4, 'Attempted Administrator Privilege Gain': 5,
-                           'access to a potentially vulnerable web application': 6, 'Information Leak': 7,
-                           'Web Application Attack': 8, 'Successful Administrator Privilege Gain': 9,
-                           'Potential Corporate Privacy Violation': 10, 'Detection of a Network Scan': 11,
-                           'Not Suspicious Traffic': 12, 'Potentially Bad Traffic': 13, 'Attempted Information Leak': 14}
-
-    micro_attack_stages_codes = [x for x, _ in micro.items()]
-    micro_attack_stages = [y for _, y in micro.items()]
-
-    if suricata_summary:
-        num_categories = len(suricata_categories)
-        percentages = [[0] * len(suricata_categories) for _ in range(len(_team_alerts))]
-    else:
-        num_categories = len(micro_attack_stages)
-        percentages = [[0] * len(micro_attack_stages) for _ in range(len(_team_alerts))]
-    indices = np.arange(num_categories)    # The x locations for the groups
-    bar_width = 0.75       # The width of the bars: can also be len(x) sequence
-
-    for tid, team in enumerate(_team_alerts):
-        for alert in team:
-            if suricata_summary:
-                percentages[tid][suricata_categories[alert[6]]] += 1
-            else:
-                percentages[tid][micro_attack_stages_codes.index(alert[9])] += 1
-        for i, acat in enumerate(percentages[tid]):
-            percentages[tid][i] = acat / len(team)
-    plots = []
-    for tid, team in enumerate(_team_alerts):
-        if tid == 0:
-            plot = plt.bar(indices, percentages[tid], bar_width)
-        elif tid == 1:
-            plot = plt.bar(indices, percentages[tid], bar_width, bottom=percentages[tid - 1])
-        else:
-            index = [x for x in range(tid)]
-            bottom = np.add(percentages[0], percentages[1])
-            for i in index[2:]:
-                bottom = np.add(bottom, percentages[i]).tolist()
-            plot = plt.bar(indices, percentages[tid], bar_width, bottom=bottom)
-        plots.append(plot)
-
-        # TODO: Decide whether to put it like this or normalize over columns
-    plt.ylabel('Percentage of occurrence')
-    plt.title('Frequency of alert category')
-    if suricata_summary:
-        plt.xticks(indices, ('c0', 'c1', 'c2', 'c3', 'c4', 'c5', 'c6', 'c7', 'c8', 'c9', 'c10', 'c11', 'c12', 'c13', 'c14'))
-    else:
-        plt.xticks(indices, [x.split('.')[1] for x in micro_attack_stages], rotation='vertical')
-    plt.tick_params(axis='x', which='major', labelsize=8)
-    plt.tick_params(axis='x', which='minor', labelsize=8)
-    # plt.yticks(np.arange(0, 13000, 1000))
-    plt.legend([plot[0] for plot in plots], _team_labels)
-    plt.tight_layout()
-    plt.savefig('data_histogram-' + experiment_name + '.png')
-    # plt.show()
+    return _team_alerts, _team_labels, _team_start_times
```

Comment:
I have made `load_data` return `_team_start_times` instead of populating a global one. I think it improves readability, while the code is doing exactly the same